cpp/tests/test_odeseir.cpp

```diff
@@ -153,3 +153,42 @@ TEST(TestSeir, check_constraints_parameters)
     
     mio::set_log_level(mio::LogLevel::warn);
 }
+
+TEST(TestSeir, get_reproduction_numbers)
+{
+    double t0   = 0;
+    double tmax = 1;
+    double dt   = 0.001;
+
+    mio::oseir::Model model;
+
+    double total_population                                                                            = 10000;
+    model.populations[{mio::Index<mio::oseir::InfectionState>(mio::oseir::InfectionState::Exposed)}]   = 100;
+    model.populations[{mio::Index<mio::oseir::InfectionState>(mio::oseir::InfectionState::Infected)}]  = 100;
+    model.populations[{mio::Index<mio::oseir::InfectionState>(mio::oseir::InfectionState::Recovered)}] = 100;
+    model.populations[{mio::Index<mio::oseir::InfectionState>(mio::oseir::InfectionState::Susceptible)}] =
+        total_population -
+        model.populations[{mio::Index<mio::oseir::InfectionState>(mio::oseir::InfectionState::Exposed)}] -
+        model.populations[{mio::Index<mio::oseir::InfectionState>(mio::oseir::InfectionState::Infected)}] -
+        model.populations[{mio::Index<mio::oseir::InfectionState>(mio::oseir::InfectionState::Recovered)}];
+
+    model.parameters.set<mio::oseir::TimeExposed>(5.2);
+    model.parameters.set<mio::oseir::TimeInfected>(6);
+    model.parameters.set<mio::oseir::TransmissionProbabilityOnContact>(0.04);
+    model.parameters.get<mio::oseir::ContactPatterns>().get_baseline()(0, 0) = 10;
+
+    model.check_constraints();
+
+    Eigen::VectorXd checkReproductionNumbers(7);
+    checkReproductionNumbers << 2.328, 2.3279906878991881, 2.3279487809434576, 2.3277601483151549, 2.3269102025388899,
```

Comment:
Maybe we should find a example with more dynamic/spread values. These values dont change really much.